File: common/datasource.go
package common

// GreptimeDBConfig 结构体表示GreptimeDB的配置
type GreptimeDBConfig struct {
	Host     string `json:"host"`
	Port     int    `json:"port"`
	Database string `json:"database"`
	Username string `json:"username"`
	Password string `json:"password"`
}

type DuckdbConfig struct {
	DbFile string `json:"db_file"`
}

type DataSource struct {
	GrepTimeCfg *GreptimeDBConfig `json:"grep_time_cfg"`
	DuckdbCfg   *DuckdbConfig     `json:"duckdb_cfg"`
	PGCfg       *PGConfig         `json:"pg_cfg"`
}

type PGConfig struct {
	Host string `json:"host"`
	Port int    `json:"port"`
	User string `json:"user"`
	Pass string `json:"pass"`
	Db   string `json:"db"`
}

File: python/common/datasource.py
class GreptimeDBConfig:
    def __init__(self, host: str, port: int, database: str, username: str, password: str):
        self.host = host
        self.port = port
        self.database = database
        self.username = username
        self.password = password


class DuckdbConfig:
    def __init__(self, db_file: str):
        self.db_file = db_file


class DataSource:
    def __init__(self, grep_time_cfg: GreptimeDBConfig, duckdb_cfg: DuckdbConfig, pg_cfg: 'PGConfig'):
        self.grep_time_cfg = grep_time_cfg
        self.duckdb_cfg = duckdb_cfg
        self.pg_cfg = pg_cfg


class PGConfig:
    def __init__(self, host: str, port: int, user: str, password: str, db: str):
        self.host = host
        self.port = port
        self.user = user
        self.password = password
        self.db = db
